File: backend/src/main/java/com/mulesoft/arc/arcdatastore/backend/models/ArcAnalyticsWeeklySessions.java
package com.mulesoft.arc.arcdatastore.backend.models;

/**
 * A class representing Daily Sessions type.
 */

public class ArcAnalyticsWeeklySessions extends ArcAnalyticsRangeResult {

    public ArcAnalyticsWeeklySessions(String day) {
        super("ArcAnalytics#WeeklySessions", day);
    }
}

File: models/arc-analytics-weekly-sessions.js
'use strict';

const {ArcAnalyticsRangeResult} = require('./arc-analytics-range-result');

/**
 * A class representing weekly Sessions type.
 */
class ArcAnalyticsWeeklySessions extends ArcAnalyticsRangeResult {
  /**
   * @param {String} startDay The start day of the date range.
   * @param {String} endDay The last day of the date range.
   * @param {Number} result The result of computation.
   */
  constructor(startDay, endDay, result) {
    super('ArcAnalytics#WeeklySessions', startDay, endDay, result);
  }
}

module.exports.ArcAnalyticsWeeklySessions = ArcAnalyticsWeeklySessions;
